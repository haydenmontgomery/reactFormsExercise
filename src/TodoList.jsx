import React, {useState} from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";
import { v4 as uuidv4 } from 'uuid';

const TodoList = () => {
  const INITIAL_STATE = [];
  const [todos, setTodos] = useState(INITIAL_STATE);
  const addTodo = (newTodo) => {
    setTodos(todos => [...todos, {...newTodo, id: uuidv4() }]);
  };

  const deleteTodo = idx => {
    setTodos(todos => (
      todos.filter((todo, i) => i !== idx)
    ));
  };

  return (
    <div className="TodoList">
      <NewTodoForm addTodo={addTodo} />
      <br />
      <ul>
        {todos.map(({ task }, idx) => 
        <Todo 
        deleteTodo={deleteTodo}
        key={idx} 
        idx={idx}
        task={task}
        />)}
      </ul>
    </div>
  )
}

export default TodoList;