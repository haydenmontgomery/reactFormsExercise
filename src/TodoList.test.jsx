import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";
import { it, expect } from "vitest";

it("renders without crashing", function () {
  render(<TodoList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add a new todo", function () {
  const { queryByText, getByLabelText } = render(<TodoList />);
  const input = getByLabelText("Task:")
  const addBtn = queryByText("Create Todo");
  expect(queryByText('Test the App')).not.toBeInTheDocument();
  fireEvent.change(input, {target: { value: "Test the App" } })
  fireEvent.click(addBtn);
  expect(queryByText('Test the App')).toBeInTheDocument();
});

it("should add a new todo and then delete it", function () {
  const { queryByText, getByLabelText } = render(<TodoList />);
  const input = getByLabelText("Task:")
  const addBtn = queryByText("Create Todo");
  expect(queryByText('Test the App')).not.toBeInTheDocument();
  fireEvent.change(input, {target: { value: "Test the App" } })
  fireEvent.click(addBtn);
  expect(queryByText('Test the App')).toBeInTheDocument();
  const deleteBtn = queryByText("X");
  fireEvent.click(deleteBtn);
  expect(queryByText('Test the App')).not.toBeInTheDocument();
});