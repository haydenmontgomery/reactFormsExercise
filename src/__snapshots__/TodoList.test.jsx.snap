// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <form
      class="NewFormTodo"
    >
      <label
        for="task"
      >
        Task: 
      </label>
      <input
        id="task"
        name="task"
        placeholder="Task"
        type="text"
        value=""
      />
      <button>
        Create Todo
      </button>
    </form>
    <br />
    <ul />
  </div>
</DocumentFragment>
`;
